pkg/quota/evaluator/core/pods.go

```diff
@@ -33,8 +34,40 @@ import (
 	"k8s.io/kubernetes/pkg/util/validation/field"
 )
 
+// listPodsByNamespaceFuncUsingInformer returns a pod listing function based on the shared informer factory.
+func listPodsByNamespaceFuncUsingInformer(f informers.SharedInformerFactory) generic.ListFuncByNamespace {
+	podLister := f.Pods().Lister()
+	return func(namespace string, options api.ListOptions) ([]runtime.Object, error) {
+		pods, err := podLister.Pods(namespace).List(options.LabelSelector)
+		if err != nil {
+			return nil, err
+		}
+		results := []runtime.Object{}
+		for i := range pods {
+			results = append(results, pods[i])
+		}
+		return results, nil
+	}
+}
+
+// listPodsByNamespaceFuncUsingClient returns a pod listing function based on the provided client.
+func listPodsByNamespaceFuncUsingClient(kubeClient clientset.Interface) generic.ListFuncByNamespace {
+	return func(namespace string, options api.ListOptions) ([]runtime.Object, error) {
```

Comment:
I looked for a way to write this generically, and spoke with @smarterclayton on it.  it's possible i am missing something, but he had directed me down this path.  Is there something obvious I am missing?